bin/prepare_multiqc.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import csv
+import argparse
+
+
+def find_file_in_subfolders(filename, root_folder='.'):
+    for root, dirs, files in os.walk(root_folder):
+        if filename in files:
+            return os.path.join(root, filename)
+    return None
+
+
+def process_matrix_file(input_file, output_file):
+
+    # Find the input file in all subfolders
+    input_file_path = find_file_in_subfolders(input_file)
+    if not input_file_path:
+        print(f"File {input_file} not found in any subfolder.")
+        sys.exit(1)
+
+
+    # Read the TSV file into a list of lists
+    with open(input_file_path, 'r') as file:
+        data = file.readlines()
+
+    # Extract row labels and matrix values
+    row_labels = []
+    matrix_values = []
+
+    for row in data:
+        values = row.strip().split('\t')
+        row_labels.append(values[0])
+        matrix_values.append(values[1:])
+
+    # Initialize the matrix
+    n = len(matrix_values)  # Number of rows/columns
+    matrix = [['' for _ in range(n)] for _ in range(n)]
+
+    # Fill the matrix
+    for i, row in enumerate(matrix_values):
+        for j, value in enumerate(row):
+            if value:  # Skip empty values
+                matrix[i][j] = value
+                matrix[j][i] = value  # Ensure symmetry
+
+    # Write the symmetric matrix to a new TSV file
+    with open(output_file, 'w') as file:
+        file.write('\t' + '\t'.join(row_labels) + '\n')
+        for i, row in enumerate(matrix):
+            file.write(row_labels[i] + '\t' + '\t'.join(row) + '\n')
+
```

Comment:
This function was not working so I had to restart from the file shared here https://github.com/abhi18av-phd-projects/mtbseq-nf-publication-analysis/pull/1 (from 2 months ago).